frameit/frames_generator/Rakefile

```diff
@@ -79,4 +100,36 @@ def convert_image_resource(path)
   image.write(path)
 end
 
+def measure_slot(path)
+  tmp_output = Time.now.to_i.to_s + ".png"
+
+  convert = MiniMagick::Tool::Convert.new
+  convert << path
+  convert.negate
+  convert.define("connected-components:verbose=true")
+  convert.define("connected-components:area-threshold=100")
+  convert.connected_components("8")
+  convert.auto_level
+  convert << tmp_output
+  command_output = convert.call
+
+  sh("rm #{tmp_output}", verbose: false)
+
+  raw_dimensions = command_output.split("\n")[1].strip
+  dimensions = /(\d+)x(\d+)\+(\d+)\+(\d+)/.match(raw_dimensions)
+  x = dimensions[3]
+  y = dimensions[4]
+  width = dimensions[1]
+  height = dimensions[2]
+  return x.to_i, y.to_i, width.to_i, height.to_i
+end
+
+def sanitize_device_name(filename)
+  basename = File.basename(filename, File.extname(filename))
+  basename = basename.gsub("Apple", "")
+  basename = basename.gsub("-", " ")
+  DEVICE_COLORS.each { |color| basename.slice!(color) }
```

Comment:
Hm, does this check somehow that all the dimensions for all colors are identical?

Reply:
No, it does not. If there were multiple images of the same device type with different dimensions, we would set the dimensions from the last processed image. 

We can add the check you are talking about, however how we should handle it? 🤔 
Shall we print the warning on the console (yellow text)? 
```bash
$ The dimensions for "iPhone 5c " are inconsistent: 640x1136+67+239, 640x1137+66+237.  
```
Or maybe we should even fail the script in that case? WDYT @janpio ?

Reply:
Hm, this is a script only a contributor would use, so no need to be super descriptive. Failing could mean that a new release of frames becomes impossible, which is not what we want. 

So I think outputting a big fat warning that list the different affected files and there offsets would be great, then the "user" can decide if they want to keep the value written into the file or overwrite it with another one (e.g. the one that applies for black+white, or the most common one).

Reply:
Cool, I see 👍 please let me know if that looks good to you:

```
Wrote the slot offsets to offsets.json
⚠️There are some affected files which have the "slot" offsets inconsistency: # written in yellow
Apple iPhone 8 Gold.png +61+219, 750
Apple iPhone 8 Silver.png +61+221, 750
Apple iPhone 8 Space Gray +61+219, 750
...
💡You can keep the offsets written into the "offsets.json" file or overwrite it with other values. 
```
From a technical point of view, I guess in a case of inconsistency we should print the whole family of devices under the same model, i.e. iPhone 8 Gold, Silver, Space Gray, etc.

WDYT? 🤔

Reply:
Perfect.

Let's just change the text a bit:

`The offsets for #{name_of_device} are not identical for all colors:`

`You can either keep the value in `offset.json` or replace it with a more appropriate value from above.

Reply:
@janpio I have made the changes 🙏 👀 

![image](https://user-images.githubusercontent.com/10795657/52462678-598b1a80-2b74-11e9-8184-e00b898b01b7.png)

By the way, there's so many logs on the console while generating the frames:

```
...
rm -rf output
mkdir -p output/1549606566
Updating and trimming image file 'output/1549606566/Apple iPad Air 2 Silver.png'
cp raw/Tablets/Apple iPad Mini 4/Device/Apple iPad Mini 4 Gold.png output/1549606566
Updating and trimming image file 'output/1549606566/Apple iPad Mini 4 Gold.png'
cp raw/Tablets/Apple iPad Mini 4/Device/Apple iPad Mini 4 Silver.png output/1549606566
Updating and trimming image file 'output/1549606566/Apple iPad Mini 4 Silver.png'
cp raw/Tablets/Apple iPad Mini 4/Device/Apple iPad Mini 4 Space Gray.png output/1549606566
Updating and trimming image file 'output/1549606566/Apple iPad Mini 4 Space Gray.png'
Wrote names of available files to files.json
Wrote the current time stamp to version.txt
Wrote the slot offsets to offsets.json
```

Maybe it's worth to check if we could simplify them somehow (more informative messages, getting rid of some) in a separate issue? 🤔 WDYT?

Reply:
Not worth the effort, as the frames are updated only every few months when Apple releases new devices.

But: You can add an empty line before and after this important message so it pops more and it is clear that something can/should be done. Can we maybe even show it in red as an error?

Reply:
<img width="891" alt="screenshot 2019-02-08 at 19 39 19" src="https://user-images.githubusercontent.com/10795657/52498904-63476900-2bda-11e9-845c-9370e9b25e50.png">

Done @janpio 💪